src/Microsoft.Framework.Caching.SqlServer/SqlServerCache.cs

```diff
@@ -0,0 +1,315 @@
+                command.Parameters.AddWithValue(SessionId, key);
+                command.Parameters.AddWithValue("UtcNow", utcNow);
+
+                await connection.OpenAsync();
+
+                var sqlDataReader = await command.ExecuteReaderAsync(CommandBehavior.SingleRow);
+
+                DateTimeOffset oldExpirationTimeUTC;
+                if (await sqlDataReader.ReadAsync())
+                {
+                    //TODO: GetFieldValueAsync
+                    oldExpirationTimeUTC = DateTimeOffset.Parse(sqlDataReader[ExpiresAtTimeUTC]?.ToString());
+                    sessionvalue = (byte[])sqlDataReader[SessionValue];
+                }
+                else
+                {
+                    return null;
+                }
+
+                // Check if the session is going to expire within the next timeout period(ex:20 minutes),
+                // then extend the expiration time.
+                if (utcNow >= oldExpirationTimeUTC.AddSeconds(-1 * _options.IdleTimeout.TotalSeconds))
+                {
+                    sqlDataReader.Dispose();
+
+                    command = new SqlCommand(SqlQueries.UpdateSessionExpiration, connection);
+                    command.Parameters.AddWithValue(SessionId, key);
+                    command.Parameters.AddWithValue(ExpiresAtTimeUTC, GetNewExpirationTime(utcNow));
+
+                    var recordsAffected = await command.ExecuteNonQueryAsync();
+                }
+            }
+
+            StartScanForExpiredSessions();
+
+            return sessionvalue;
+        }
+
+        public void Refresh([NotNull] string key)
+        {
+            var value = Get(key);
+        }
+
+        public async Task RefreshAsync([NotNull] string key)
+        {
+            var value = await GetAsync(key);
+        }
+
+        public void Remove([NotNull] string key)
+        {
+            using (var connection = new SqlConnection(_options.ConnectionString))
+            {
+                var command = new SqlCommand(SqlQueries.DeleteSession, connection);
+                command.Parameters.AddWithValue(SessionId, key);
+
+                connection.Open();
+
+                var effectedRowCount = command.ExecuteNonQuery();
+            }
+
+            StartScanForExpiredSessions();
+        }
+
+        public async Task RemoveAsync([NotNull] string key)
+        {
+            using (var connection = new SqlConnection(_options.ConnectionString))
+            {
+                var command = new SqlCommand(SqlQueries.DeleteSession, connection);
+                command.Parameters.AddWithValue(SessionId, key);
+
+                await connection.OpenAsync();
+
+                var effectedRowCount = await command.ExecuteNonQueryAsync();
+            }
+
+            StartScanForExpiredSessions();
+        }
+
+        public void Set([NotNull] string key, [NotNull] byte[] value, DistributedCacheEntryOptions options)
```

Comment:
yeah :+1:  ...(probably this is something related to my earlier discussion about creating just a 'store' for session and not get tied up with caching...this does not feel intuitive intuitive to me actually...like saying `AddSqlServerCache` and then `AddSession`)